libraries/KVStore/KVStore.h

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_KVSTORE_H
+#define MBED_KVSTORE_H
+
+#include <stdint.h>
+#include <string.h>
+
+// FIXME define mbed errors adequately
```

Comment:
What is the problem with the following defines? or the comment is outdated ?

Reply:
I forgot about that. The point is that mbedos has a particular way of defining errors and make them reusable and work across all the different modules it has. I hardcoded the ones we care about for kvstore. 

Do we care about having the complete definition of error codes? or should we just use the hardcoded ones?

Reply:
We can go with the hardcoded ones, but we have to check there are no clashes with the other imported mbed modules like the ones in the `BlockDevices` and `Storage` folders